gremlin-go/driver/resultSet.go

```diff
@@ -109,6 +144,7 @@ func (channelResultSet *channelResultSet) All() []*Result {
 }
 
 func (channelResultSet *channelResultSet) addResult(r *Result) {
+	channelResultSet.channelMux.Lock()
```

Comment:
Q: Do we want to check if the channel is closed here before pushing onto it since we grabbed the channelMux?

Reply:
I don't think so. I am fine without checking because it would indicate a much larger protocol violation if this were to happen and I am okay with leaving that for us to figure out in the protocol.